admin_app/src/components/LanguageButtonBar.tsx

```diff
@@ -1,95 +1,152 @@
+  const isLanguageEnabled = (languageId: number) => {
+    return typeof enabledLanguages === 'undefined' || enabledLanguages.includes(languageId);
+  }
+  React.useEffect(() => {
+    const fetchLanguages = async () => {
+      try {
+        const languages = await apiCalls.getLanguageList(token!);
+        setLangList(languages);
+
+        const defaultLanguage = langList
+          .find(lang => lang.language_id === defaultLanguageId);
+        if (defaultLanguage && !selectedLang) {
+          setSelectedLang(defaultLanguageId);
+          onLanguageSelect(defaultLanguageId);
+        }
+      } catch (error) {
+        console.error('Failed to fetch languages:', error);
+      }
+    };
+
+    fetchLanguages();
+
+
+  }, [token, expandable, defaultLanguageId]);
+
+  const handleMenuClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleMenuClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleToggleButtonSelect = (language: Language) => {
+    setSelectedLang(language.language_id);
+    onLanguageSelect(language.language_id);
+  };
+  const handleMenuItemSelect = (language: Language) => {
+    setSelectedLang(language.language_id);
+    setAnchorEl(null);
+    onMenuItemSelect && onMenuItemSelect(language.language_id);
+  }; return (
     <Layout.FlexBox
       flexDirection={"row"}
       gap={sizes.smallGap}
       {...appStyles.alignItemsCenter}
     >
       <ToggleButtonGroup>
-        {langList.map(
-          (lang, index) =>
+        {langList
+          .filter(lang => !isEdit || isLanguageEnabled(lang.language_id))
+          .map((lang) =>
             lang && (
+
               <ToggleButton
-                key={lang.code}
-                value={lang.code}
+                key={lang.language_id}
+                value={lang.language_id}
                 size="medium"
+                onClick={() => handleToggleButtonSelect(lang)}
+                disabled={!isEdit && !isLanguageEnabled(lang.language_id)}
                 sx={[
                   {
                     border: 0,
                     borderBottomColor: appColors.outline,
                     borderBottomWidth: 1,
                   },
-                  selectedLang === lang?.code && {
+                  selectedLang === lang?.language_id && {
                     borderBottomColor: appColors.primary,
                     borderBottomWidth: 3,
                     color: appColors.primary,
                   },
                 ]}
-                onClick={() => setSelectedLang(lang.code)}
               >
                 <Layout.Spacer horizontal multiplier={0.5} />
-                {lang?.label}
+                {lang?.language_name}
                 <Layout.Spacer horizontal multiplier={0.5} />
               </ToggleButton>
             ),
-        )}
+          )
+        }
       </ToggleButtonGroup>
-      {expandable && LANGUAGE_OPTIONS.length > langList.length && (
```

Comment:
Any reason you took this type of login out? I like the idea that you don't need this circle if there are no languages left to add.

Reply:
I will add it back in next PR.